src/panels/lovelace/entity-rows/hui-input-select-entity-row.ts

```diff
@@ -0,0 +1,110 @@
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { repeat } from "lit-html/directives/repeat";
+import { TemplateResult } from "lit-html";
+import "@polymer/paper-dropdown-menu/paper-dropdown-menu";
+import "@polymer/paper-item/paper-item";
+import "@polymer/paper-listbox/paper-listbox";
+
+import "../../../components/entity/state-badge";
+import "./hui-error-entity-row";
+
+import computeStateName from "../../../common/entity/compute_state_name";
+import { HomeAssistant } from "../../../types";
+import { EntityRow, EntityConfig } from "./types";
+
+class HuiInputSelectEntityRow extends LitElement implements EntityRow {
+  public hass?: HomeAssistant;
+  private _config?: EntityConfig;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public setConfig(config: EntityConfig): void {
+    if (!config || !config.entity) {
+      throw new Error("Invalid Configuration: 'entity' required");
+    }
+
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+
+    if (!stateObj) {
+      return html`
+        <hui-error-entity-row
+          .entity="${this._config.entity}"
+        ></hui-error-entity-row>
+      `;
+    }
+
+    return html`
+      ${this.renderStyle()}
+      <state-badge .stateObj="${stateObj}"></state-badge>
+      <paper-dropdown-menu
+        label="${this._config.name || computeStateName(stateObj)}"
+      >
+        <paper-listbox
+          slot="dropdown-content"
+          selected="${stateObj.attributes.options.indexOf(stateObj.state)}"
+        >
+          ${
+            repeat(
+              stateObj.attributes.options,
+              (option) =>
+                html`
+                  <paper-item
+                    @click="${() => this._selectedChanged(option as string)}"
```

Comment:
Don't do this. Instead mimic how the old component behaved: observing `selected-item-label` (as seen by it using `{…}` which indicates 2 way databinding). You can do this by listening to the `selected-item-label-changed` event. Don't attach individual listeners to each item.

Reply:
Hopefully I didn't botch that too much with the commit I just pushed (it works) :) I was pretty unsure of what I was doing with this listener.